app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/securities/ConfirmFullRepaymentController.scala

```diff
@@ -16,18 +16,145 @@
 import com.google.inject.Inject
 import com.google.inject.Singleton
+import play.api.data.Form
+import play.api.mvc.Action
+import play.api.mvc.AnyContent
+import play.api.mvc.Call
+import play.api.mvc.Request
+import play.api.mvc.Result
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ErrorHandler
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ViewConfig
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.Forms.confirmFullRepaymentForm
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.mixins.WorkInProgressMixin
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.{routes => baseRoutes}
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.SecuritiesJourney
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.SecuritiesJourney.Checks.{
+  hasMRNAndDisplayDeclarationAndRfS,
+  canContinueTheClaimWithChoosenRfS,
+  declarantOrImporterEoriMatchesUserOrHasBeenVerified
+}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.answers.YesNo
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.answers.YesNo.No
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.answers.YesNo.Yes
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.securities.confirm_full_repayment
 
 import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Try
 
 @Singleton
 class ConfirmFullRepaymentController @Inject() (
-  val jcc: JourneyControllerComponents
-)(implicit viewConfig: ViewConfig, ec: ExecutionContext)
+  val jcc: JourneyControllerComponents,
+  confirmFullRepaymentPage: confirm_full_repayment
+)(implicit viewConfig: ViewConfig, errorHandler: ErrorHandler, ec: ExecutionContext)
     extends SecuritiesJourneyBaseController
-    with WorkInProgressMixin[SecuritiesJourney]
+    with SecuritiesJourneyRouter
+    with Logging {
+
+  private val form: Form[YesNo] = confirmFullRepaymentForm
+
+  // todo import SecuritiesJourney.Checks._
+
+  // Allow actions only if the MRN, RfS and ACC14 declaration are in place, and the EORI has been verified.
+  override val actionPrecondition: Option[Validate[SecuritiesJourney]] =
+    Some(
+      hasMRNAndDisplayDeclarationAndRfS &
+        canContinueTheClaimWithChoosenRfS &
+        declarantOrImporterEoriMatchesUserOrHasBeenVerified
+    )
+
+  // GET          /securities/confirm-full-repayment
+  // @uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.ConfirmFullRepaymentController.showFirst()
+  // @(form: Form[YesNo], securityId: String, totalValue: BigDecimal, postAction: Call)
+  //    (implicit request: RequestWithSessionData[_], messages: Messages, viewConfig: ViewConfig)
+  def showFirst(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    journey.getSecurityDepositIds.headOption.fold(
```

Comment:
better steer user back to /chec-declaration-details

Reply:
Done